droidlet/lowlevel/hello_robot/remote/remote_hello_robot_ros.py

```diff
@@ -0,0 +1,232 @@
+
+    def __init__(self, ip):
+        self._ip = ip
+        self._robot = Robot()
+        self._robot.start()
+        self._done = True
+        self.cam = None
+        # Read battery maintenance guide https://docs.hello-robot.com/battery_maintenance_guide/
+        self._load_urdf()
+        self.tilt_correction = 0.0
+
+    def _load_urdf(self):
+        import os
+
+        urdf_path = os.path.join(
+            os.getenv("HELLO_FLEET_PATH"),
+            os.getenv("HELLO_FLEET_ID"),
+            "exported_urdf",
+            "stretch.urdf",
+        )
+
+        from pytransform3d.urdf import UrdfTransformManager
+        import pytransform3d.transformations as pt
+        import pytransform3d.visualizer as pv
+        import numpy as np
+
+        self.tm = UrdfTransformManager()
+        with open(urdf_path, "r") as f:
+            urdf = f.read()
+            self.tm.load_urdf(urdf)
+
+    def set_tilt_correction(self, angle):
+        """
+        angle in radians
+        """
+        print(
+            "[hello-robot] Setting tilt correction " "to angle: {} degrees".format(degrees(angle))
+        )
+
+        self.tilt_correction = angle
+
+    def get_camera_transform(self):
+        head_pan = self.get_pan()
+        head_tilt = self.get_tilt()
+
+        if self.tilt_correction != 0.0:
+            head_tilt += self.tilt_correction
+
+        # Get Camera transform
+        self.tm.set_joint("joint_head_pan", head_pan)
+        self.tm.set_joint("joint_head_tilt", head_tilt)
+        camera_transform = self.tm.get_transform("camera_color_frame", "base_link")
+
+        # correct for base_link's z offset from the ground
+        # at 0, the correction is -0.091491526943
+        # at 90, the correction is +0.11526719 + -0.091491526943
+        # linear interpolate the correction of 0.023775
+        interp_correction = 0.11526719 * abs(head_tilt) / radians(90)
+        # print('interp_correction', interp_correction)
+
+        camera_transform[2, 3] += -0.091491526943 + interp_correction
+
+        return camera_transform
+
+    def get_base_state(self):
+        # Best (from SLAM)
+        return self._robot.get_slam_pose()
+        # second Best (from lidar scanning)
+        # return self._robot.get_scan_matched_pose()
+        # Worst, from wheel encoder only
+        # return self._robot.get_odom()
+
+    def get_slam_pose(self):
+        return self._robot.get_slam_pose()
+
+    def pull_status(self):
+        pass
+
+    def is_base_moving(self):
+        moving = self._robot.is_moving()
+        print("is_moving", moving)
+        return moving
+
+    def get_pan(self):
+        return self._robot.get_joint_state("head_pan")
+
+    def get_tilt(self):
+        return self._robot.get_joint_state("head_tilt")
+
+    def set_pan(self, pan):
+        self._robot.send_command("joint_head_pan", pan)
+
+    def set_tilt(self, tilt):
+        self._robot.send_command("joint_head_tilt", tilt)
+
+    def reset_camera(self):
```

Comment:
description - resets camera to home position?

Reply:
done